### PR TITLE
Gauges: add centerTemplate property

### PR DESCRIPTION
**Pull request by Krijovnick**

### CODE REVIEW

File: js/viz/core/center_template.js [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+
+            const center = this._getCenter();
+
+            this._getTemplate(template).render({
+                model: this,
+                container: centerTemplateGroup.element,
+                onRendered: () => {
+                    const group = centerTemplateGroup;
+                    const bBox = group.getBBox();
+                    group.move(center.x - (bBox.x + bBox.width / 2), center.y - (bBox.y + bBox.height / 2));
```
Comment (ksercs) [inline js/viz/core/center_template.js]: Let's move part of calculations to separate variables to simplify this row?
```suggestion
                    const bBoxCenterX = bBox.x + bBox.width / 2;
                    const bBoxCenterY = bBox.y + bBox.height / 2;
                    group.move(center.x - bBoxCenterX, center.y - bBoxCenterY);
```

File: js/viz/core/center_template.js [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+    init: noop,
+    dispose: function() {
+        this._centerTemplateGroup.linkOff().dispose();
+    },
+    extenders: {
+        _createHtmlStructure() {
+            this._centerTemplateGroup = this._renderer.g()
+                .attr({ class: 'dxc-hole-template' })
+                .linkOn(this._renderer.root, 'center-template')
+                .css(patchFontOptions(this._themeManager._font))
```
Comment (ksercs) [inline js/viz/core/center_template.js]: It's a good idea to move a parameter function call to a separate variable to simplify a debug.
When u debug u will see which parameter u pass before this chaining and can debug parameter getter separately.
```suggestion
            const patchedFontOptions = patchFontOptions(this._themeManager._font);
            this._centerTemplateGroup = this._renderer.g()
                ...
                .css(patchedFontOptions)
```

File: testing/tests/DevExpress.viz.gauges/barGauge_new.tests.js [hunk 3]
```diff
@@ -8,6 +8,23 @@ import loadingIndicatorModule from 'viz/core/loading_indicator';
...
+        sinon.stub(rendererModule, 'Renderer', () => {
+            return this.renderer;
+        });
+    },
+    afterEach() {
+        rendererModule.Renderer.restore();
+    },
+
+    createGauge(options) {
+        return $('<div>').appendTo($('#qunit-fixture')).dxBarGauge(options).dxBarGauge('instance');
```
Comment (ksercs) [inline testing/tests/DevExpress.viz.gauges/barGauge_new.tests.js]: It's just easier to read
```suggestion
        return $('<div>')
            .appendTo($('#qunit-fixture'))
            .dxBarGauge(options)
            .dxBarGauge('instance');
```

**Review by ksercs — DISMISSED** [review]
Let's postpone a question with using of a mixin

**Review by alexanderPolosatov — DISMISSED** [review]

File: js/viz/bar_gauge.d.ts [hunk 4]
```diff
@@ -174,6 +174,13 @@ export interface dxBarGaugeOptions extends BaseWidgetOptions<dxBarGauge> {
        */
       startAngle?: number;
     };
+    /**
+     * @docid
+     * @default undefined
+     * @type_function_return string|SVGElement|jQuery
+     * @public
+     */
+    centerTemplate?: template | ((component: dxBarGauge, element: SVGGElement) => string | UserDefinedElement<SVGElement>);
```
Comment (IlyaKhD) [inline js/viz/bar_gauge.d.ts:184]: @OlegKipchatov, please remind - do we need something special here for templates? `@type template` or like?

File: js/viz/circular_gauge.d.ts [hunk 5]
```diff
@@ -76,6 +84,13 @@ export interface dxCircularGaugeOptions extends BaseGaugeOptions<dxCircularGauge
        */
       startAngle?: number;
     };
+    /**
+     * @docid
+     * @default undefined
+     * @type_function_return string|SVGElement|jQuery
+     * @public
+     */
+    centerTemplate?: template | ((component: dxCircularGauge, element: SVGGElement) => string | UserDefinedElement<SVGElement>);
```
Comment (IlyaKhD) [inline js/viz/circular_gauge.d.ts:94]: @OlegKipchatov, same here

**Review by IlyaKhD — DISMISSED** [review]

**Review by OlegKipchatov — APPROVED** [review]

**Review by ksercs — APPROVED** [review]

**Review by vladaskorohodova — APPROVED** [review]

**Review by alexanderPolosatov — APPROVED** [review]

**Review by EugeniyKiyashko — APPROVED** [review]

**Review by IlyaKhD — APPROVED** [review]

**Pull request merged**